### PR TITLE
Feat/deletecard

### PR DESCRIPTION
**Pull request by PhilippWeiss85**
What have been done?
- added delete icon + styling
- added dynamic colors for card headlines depending on match/training
- removed the gameplan page for now because the whole functionality of the app is rendered on the form page
- added delete card function

user story: #8

### CODE REVIEW

File: components/DeleteButton.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  return <StyledButton>{children}</StyledButton>;
+}
+
+const StyledButton = styled.button`
+  padding: 0.2em 0.5em;
+  margin: 0 1em;
+  background-color: transparent;
+  color: var(--text-primary);
+  cursor: pointer;
+  text-decoration: none;
```
Comment (philmetscher) [inline components/DeleteButton.js]: a button has no text-decoration as standard, i think 🤔

File: components/DeleteButton.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  font-size: 1.5em;
+  border: none;
+
+  &:hover {
+    transition: 0.2s ease;
+    filter: drop-shadow(5px 4px 4px red);
```
Comment (philmetscher) [inline components/DeleteButton.js]: never use colors like `red`. Use Hex-Colors or RGB or HSL.
Colors like `red` can be interpreted differently by different browsers

Pls look inside all styled Components and correct this

File: pages/form/index.js [hunk 3]
```diff
@@ -2,8 +2,38 @@ import GameCardForm from "../../components/Form/GameCardForm";
 import { useState } from "react";
 import GameCard from "../../components/GameCard/GameCard";
 
+const games = [
+  {
+    id: Math.random().toString(20).substring(3),
```
Comment (philmetscher) [inline pages/form/index.js]: U could use [nanoid](https://github.com/ai/nanoid)

File: components/GameCard/GameCard.js [hunk 4]
```diff
@@ -1,16 +1,31 @@
...
+      <CardType type={type}>{type}</CardType>
       <CardContainer>
         <PlayerDiv>
           <p>
             <BsPersonCircle /> {name}
           </p>
         </PlayerDiv>
+        <DeleteButtonContainer>
+          <DeleteButton>
+            <BsX onClick={() => deleteCard(id)} />
```
Comment (philmetscher) [inline components/GameCard/GameCard.js]: click event on icon? Set it on the button

File: components/GameCard/GameCard.js [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
-export default function GameCard({ type, name, date, time, place, court }) {
+export default function GameCard({
+  type,
+  name,
+  date,
+  time,
+  place,
+  court,
+  deleteCard,
+  id,
```
Comment (philmetscher) [inline components/GameCard/GameCard.js:13]: do you need the `id`?

**Review by philmetscher — CHANGES_REQUESTED** [review]
Sorry. Some changes 😉 😜

File: components/DeleteButton.js (hunk 1 above)
Comment (Ozanyurdakul98) [reply components/DeleteButton.js]: true

**Review by Ozanyurdakul98 — COMMENTED** [review]
Ich habe zu Phil nichts hinzuzufügen. Gebe also ein approve

File: components/DeleteButton.js (hunk 1 above)
Comment (PhilippWeiss85) [reply components/DeleteButton.js]: that was just a lazy copy&paste error from my former styledlink component

File: components/GameCard/GameCard.js (hunk 4 above)
Comment (PhilippWeiss85) [reply components/GameCard/GameCard.js]: Hey, unfortunately onClick is not working when set on the deletebutton styled component... tough I did set is on the DeleteButtonContainer.

File: components/GameCard/GameCard.js (hunk 5 above)
Comment (PhilippWeiss85) [reply components/GameCard/GameCard.js:13]: I think so... how the id prop is used by the onClick on the deleteCardContainer

**Review by Ozanyurdakul98 — APPROVED** [review]

File: components/GameCard/GameCard.js [hunk 6]
```diff
@@ -21,9 +21,9 @@ export default function GameCard({
             <BsPersonCircle /> {name}
           </p>
         </PlayerDiv>
-        <DeleteButtonContainer>
+        <DeleteButtonContainer onClick={() => deleteCard(id)}>
           <DeleteButton>
-            <BsX onClick={() => deleteCard(id)} />
+            <BsX />
           </DeleteButton>
         </DeleteButtonContainer>
```
Comment (philmetscher) [inline components/GameCard/GameCard.js:28]: you should set it on the `DeleteButton`. You can't set it directly via
**components/GameCard/GameCard.js**
```javascript
<DeleteButton handleClick={() => deleteCard(id)}>
  <BsX />
</DeleteButton>
```

**components/DeleteButton.js**
```javascript
export default function DeleteButton({ children, handleClick }) {
  return <StyledButton onClick={() => handleClick}>{children}</StyledButton>;
}
```

(`handleClick` is just a prop. Name it whatever u want)

**Pull request merged**